state/storage.go

```diff
@@ -80,8 +80,44 @@ const (
 	StorageKindFilesystem
 )
 
+// NewStorageBackend creates a backend for managing storage.
+func NewStorageBackend(st *State) (*storageBackend, error) {
+	// TODO(wallyworld) - we should be passing in a Model not a State
+	// (but need to move stuff off State first)
+	m, err := st.Model()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	registry, err := st.storageProviderRegistry()
+	// TODO(caas) - implement storage provider registry
+	if err != nil && !errors.IsNotImplemented(err) {
```

Comment:
The registry is nil for CAAS models (doesn't matter as storage CLI disabled for CAAS models).
Next PR implements the registry so this will disappear.